plugins/native/module/vtkF3DQuakeMDLImporter.cxx

```diff
@@ -0,0 +1,536 @@
+        { return pair.first > animationIndex; }));
+    firstFrameIndex =
+      firstFrameIndex <= static_cast<int>(this->AnimationIds.size()) ? firstFrameIndex : 0;
+    lastFrameIndex = lastFrameIndex <= static_cast<int>(this->AnimationIds.size())
+      ? lastFrameIndex - 1
+      : lastFrameIndex;
+    for (int i = firstFrameIndex; i <= lastFrameIndex; i++)
+    {
+      this->ActiveFrames.erase(std::remove(this->ActiveFrames.begin(), this->ActiveFrames.end(), i),
+        this->ActiveFrames.end());
+    }
+    this->ActiveAnimationIds.erase(
+      std::remove(this->ActiveAnimationIds.begin(), this->ActiveAnimationIds.end(), animationIndex),
+      this->ActiveAnimationIds.end());
+  }
+
+  //----------------------------------------------------------------------------
+  void ImportActors(vtkRenderer* renderer)
+  {
+    vtkNew<vtkActor> actor;
+    vtkNew<vtkPolyDataMapper> mapper;
+    mapper->SetInputData(Mesh[0]);
+    actor->SetMapper(mapper);
+    actor->SetTexture(Texture);
+    renderer->AddActor(actor);
+    renderer->SetBackground(0, 0, 0);
+    this->Mapper = mapper;
+  }
+
+  //----------------------------------------------------------------------------
+  void SetFrameRate(double frameRate)
+  {
+    this->FrameRate = frameRate;
+  }
+
+  //----------------------------------------------------------------------------
+  void GetTimeRange(vtkIdType vtkNotUsed(animationIndex), double timeRange[2])
+  {
+    timeRange[0] = 0.0;
+    timeRange[1] = (1.0 / this->FrameRate) * this->AnimationIds.size();
+  }
+
+  vtkF3DQuakeMDLImporter* Parent;
+  std::string Description;
+  vtkSmartPointer<vtkPolyDataMapper> Mapper;
+  std::vector<vtkSmartPointer<vtkPolyData>> Mesh;
+  std::vector<std::pair<int, float>> AnimationIds;
+  std::vector<std::string> AnimationNames;
+  std::vector<int> ActiveFrames;
+  std::vector<int> ActiveAnimationIds;
+  int NumberOfAnimations = 0;
+  double FrameRate = 60.0;
+  vtkSmartPointer<vtkTexture> Texture;
+};
+
+//----------------------------------------------------------------------------
+vtkF3DQuakeMDLImporter::vtkF3DQuakeMDLImporter()
+  : Internals(new vtkF3DQuakeMDLImporter::vtkInternals(this)) {
+
+  };
+
+//----------------------------------------------------------------------------
+int vtkF3DQuakeMDLImporter::ImportBegin()
+{
+  return this->Internals->ReadScene(this->FileName);
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DQuakeMDLImporter::ImportActors(vtkRenderer* renderer)
+{
+  this->Internals->ImportActors(renderer);
+}
+
+//----------------------------------------------------------------------------
+std::string vtkF3DQuakeMDLImporter::GetOutputsDescription()
+{
+  return this->Internals->Description;
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DQuakeMDLImporter::UpdateTimeStep(double timeValue)
+{
+  return this->Internals->UpdateTimeStep(timeValue);
```

Comment:
It seems like animation isn't covered yet. Can we try to find a .mdl file containing animations?